pkg/nettest/nettest.go

```diff
@@ -0,0 +1,68 @@
+// Package nettest provides functionality to run tests that access the public network.
+package nettest
+
+import (
+	"errors"
+	"net"
+	"os"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+)
+
+func lock(lockFilePath string) {
+	for {
+		outFile, err := os.OpenFile(lockFilePath, os.O_CREATE|os.O_EXCL, 0o600)
+		if err == nil {
+			outFile.Close()
+			return
+		}
+		time.Sleep(5 * time.Second)
+	}
+}
+
+func unlock(lockFilePath string) {
+	os.Remove(lockFilePath)
+}
+
+func runSerial(t *testing.T, lockFilePath string) {
+	lock(lockFilePath)
+	t.Logf("network test start: %s", t.Name())
+
+	t.Cleanup(func() {
+		t.Logf("network test done: %s", t.Name())
+		unlock(lockFilePath)
+	})
+}
+
+// Run can be used by test that access the public network.
+func Run(t *testing.T) {
+	if os.Getenv("NETWORK_TEST") == "" {
+		t.Skip("set environment variable NETWORK_TEST=1 to run")
+		return
+	}
+
+	if runtime.GOOS != "darwin" && runtime.GOOS != "linux" {
+		return
+	}
+
+	if lockFilePath := os.Getenv("TEST_LOCK_FILE_PATH"); lockFilePath != "" {
+		runSerial(t, lockFilePath)
+	}
+}
+
+// Retryable returns whether the error warrants a retry.
+func Retryable(err error) bool {
+	var netErr net.Error
+	if errors.As(err, &netErr) {
+		if netErr.Temporary() || netErr.Timeout() {
+			return true
+		}
+	}
+	// https://github.com/golang/go/blob/a5d61be040ed20b5774bff1b6b578c6d393ab332/src/net/http/serve_test.go
+	if errStr := err.Error(); strings.Contains(errStr, "timeout") && strings.Contains(errStr, "TLS handshake") {
```

Comment:
Nit, but not clear how the comment is supposed to help understand the additional condition? It would help to link to a specific part of that (big) file, or (and?) add some explanation.